includes/class-llms-forms.php

```diff
@@ -0,0 +1,885 @@
+		 * NOTE: Removing core forms (as well as modifying the ids / keys) may cause areas of LifterLMS to stop working.
+		 *
+		 * @since [version]
+		 *
+		 * @param  array $locations Associative array of form location information.
+		 */
+		return apply_filters(
+			'llms_forms_get_locations',
+			array(
+				'checkout'     => array(
+					'name'        => __( 'Checkout', 'lifterlms' ),
+					'description' => __( 'Handles new user registration and existing user information updates during checkout and enrollment.', 'lifterlms' ),
+					'title'       => __( 'Billing Information', 'lifterlms' ),
+					'meta'        => array(
+						'_llms_form_show_title' => 'yes',
+						'_llms_form_is_core'    => 'yes',
+					),
+				),
+				'registration' => array(
+					'name'        => __( 'Registration', 'lifterlms' ),
+					'description' => __( 'Handles new user registration and existing user information updates for open registration on the student dashboard and wherever the [lifterlms_registration] shortcode is used.', 'lifterlms' ),
+					'title'       => __( 'Register', 'lifterlms' ),
+					'meta'        => array(
+						'_llms_form_show_title' => 'yes',
+						'_llms_form_is_core'    => 'yes',
+					),
+				),
+				'account'      => array(
+					'name'        => __( 'Account', 'lifterlms' ),
+					'description' => __( 'Handles user account information updates on the edit account area of the student dashboard.', 'lifterlms' ),
+					'title'       => __( 'Edit Account Information', 'lifterlms' ),
+					'meta'        => array(
+						'_llms_form_show_title' => 'no',
+						'_llms_form_is_core'    => 'yes',
+					),
+				),
+			)
+		);
+
+	}
+
+	/**
+	 * Retrieve the forms post type name.
+	 *
+	 * @since [version]
+	 *
+	 * @return string
+	 */
+	public function get_post_type() {
+		return $this->post_type;
+	}
+
+	/**
+	 * Retrieve a permalink for a given form post.
+	 *
+	 * This is primarily used by the Block Editor to allow quick links to the form from within the editor.
+	 *
+	 * @since [version]
+	 *
+	 * @param WP_Post $post Form post object.
+	 * @return string|false
+	 */
+	protected function get_permalink( $post ) {
+
+		$url      = false;
+		$location = get_post_meta( $post->ID, '_llms_form_location', true );
+
+		switch ( $location ) {
+
+			case 'account':
+				$url = llms_get_endpoint_url( 'edit-account', '', llms_get_page_url( 'myaccount' ) );
+				break;
+
+			case 'checkout':
+				$url = llms_get_page_url( 'checkout' );
+
+				// Add an access plan to the URL.
+				$plans = new WP_Query(
+					array(
+						'post_type'      => 'llms_access_plan',
+						'posts_per_page' => 1,
+						'orderby'        => 'ID',
+						'order'          => 'ASC',
+					)
+				);
+				if ( $plans->have_posts() ) {
+					$url = add_query_arg( 'plan', $plans->posts[0]->ID, $url );
+				}
+				break;
+
+			case 'registration':
+				if ( llms_parse_bool( get_option( 'lifterlms_enable_myaccount_registration', 'no' ) ) ) {
+					$url = llms_get_page_url( 'myaccount' );
```

Comment:
Of course you know this, but anyways...
At the moment this doesn't make a lot of sense as the account page for logged in users (as we are) won't show the registration form. I guess your idea is to, in the future, enhance the view-manager so to be able to see the account page as visitor, as we do (want to do) in #874